File: pset1/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    float n;
    do
    {
        n = get_float("digite o troco:  ");
    }
    while (n <= 0);

    int centavos = round(n * 100);
    int qd_moedas = 0;

    for (qd_moedas = 0; centavos > 0;)
        if (centavos >= 25)
        {
            centavos = centavos - 25;
            qd_moedas++;
            printf(" 25 ");
        }
        else if (centavos >= 10 && centavos < 25)
        {
            centavos = centavos - 10;
            qd_moedas++;
            printf(" 10 ");
        }
        else if (centavos >= 5 && centavos < 10)
        {
            centavos = centavos - 5;
            qd_moedas++;
            printf(" 5 ");
        }
        else if (centavos >= 1 && centavos < 5)
        {
            centavos = centavos - 1;
            qd_moedas++;
            printf(" 1 ");
        }

    printf("%d, \n", qd_moedas);
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:

    n = get_float('Change owed: ')

    if n > 0:
        break
    print('Change owed: ivalid')

cont = 0
while n > 0:


    if n >= 0.25:
        cont += 1
        n -= 0.25
        n = round(n, 2)

    elif n >= 0.10 and n < 0.25:
        cont += 1
        n -= 0.10
        n = round(n, 2)

    elif n >= 0.05 and n < 0.10:
        cont += 1
        n -= 0.05
        n = round(n, 2)

    elif n >= 0.01 and n < 0.05:
        cont += 1
        n -= 0.01
        n = round(n, 2)

print(cont)
